home/import_ordered_content_sets.py

```diff
@@ -0,0 +1,227 @@
+            time = times[idx]
+            unit = units[idx]
+            before_or_after = before_or_afters[idx]
+            contact_field = contact_fields[idx]
+            if page:
+                ordered_set.pages.append(
+                    (
+                        "pages",
+                        {
+                            "contentpage": page,
+                            "time": time or "",
+                            "unit": unit or "",
+                            "before_or_after": before_or_after or "",
+                            "contact_field": contact_field or "",
+                        },
+                    )
+                )
+            else:
+                logger.warning(f"Content page not found for slug '{page_slug}'")
+
+    def _create_ordered_set_from_row(
+        self, index: int, row: dict[str, str]
+    ) -> OrderedContentSet:
+        """
+        Create an ordered content set from a single row of the file.
+
+        :param index: The row number of the row being processed.
+        :param row: The row of the file as a dictionary.
+        :return: An instance of OrderedContentSet.
+        :raises ImportException: If time, units, before_or_afters, page_slugs and contact_fields are not all equal length.
+        """
+        ordered_set = self._get_or_init_ordered_content_set(row, row["Name"])
+
+        times = self._csv_to_list(row["Time"])
+        units = self._csv_to_list(row["Unit"])
+        before_or_afters = self._csv_to_list(row["Before Or After"])
+        page_slugs = self._csv_to_list(row["Page Slugs"])
+        # backwards compatiblilty if there's only one contact field
+        contact_fields = row["Contact Field"].split(",")
+        contact_fields = (
+            [p.strip() for p in contact_fields]
+            if len(contact_fields) > 1
+            else [contact_fields[0]] * len(times)
+        )
+
+        self._validate_extracted_values(
+            index, row, times, units, before_or_afters, page_slugs, contact_fields
+        )
+
+        self._create_pages(
+            ordered_set, times, units, before_or_afters, page_slugs, contact_fields
+        )
+
+        ordered_set.save()
+        return ordered_set
+
+    def _set_progress(self, progress: int) -> None:
+        self.progress_queue.put_nowait(progress)
+
+    def perform_import(self) -> None:
+        """
+        Import ordered content sets from a file.
+
+        This method reads the file associated with the importer, processes its
+        content based on the specified file type (XLSX or CSV), and creates
+        ordered content sets by extracting relevant information from each row.
+        The progress of the import operation is reported via a queue.
+
+        Raises:
+            ImportException: If any inconsistencies are found in the data
+            row while creating ordered content sets.
+        """
+        file = self.file.read()
+        lines = []
+        if self.filetype == "XLSX":
```

Comment:
Can we separate the CSV-specific and XLSX-specific bits into their own methods that both return a list of rows? That way we don't have to worry about the details of parsing each file type in this method.